apps/system/lockscreen/js/charging.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+(function(exports) {
+
+/**
+ * Creates an object used for refreshing the charging string UI element.
+ * @class  ChargingStatus
+ */
+function ChargingStatus() {
+  this.refreshCB = null;
+}
+
+ChargingStatus.prototype.start = function cs_start() {
+  this.refresh();
+
+  if (!this.refreshCB) {
+    this.refreshCB = this.refresh.bind(this);
+
+    navigator.battery.addEventListener('chargingchange',
+      this.refreshCB);
```

Comment:
And the most important thing is if it exactly a DOMEvent target you should write it as:

    navigator.battery.addEventListener('chargingchange', this);

And set a `handleEvent` method under this prototype, which would be automatically bound to the event, with the fixed context. The binding trick you used and I've commented is only for those (bad) APIs support no automatically context binding. For example, the `mozSettings` API.